### ISSUE TITLE
Unexpected diagnostic ID when overload resolution fails due to method group conversion referencing instance member in static context

### ISSUE THREAD

**Issue by jhinder** [dotnet/roslyn#51953]
**Version Used**: 3.10.0-1.21125.6 (VS 16.10 Preview)

**Steps to Reproduce**:

```c#
delegate void A();
delegate void B(object o);
class C
{
    static void Method(A a) { }
    static void Method(B b) { }

    void MatchesA() { }
    void MatchesB(object o) { }

    static void M()
    {
        Method(MatchesA); // CS0120
        Method(MatchesB); // CS1503
    }
}
```

**Expected Behavior**:
Both invocations report CS0120 ("An object reference is required for the non-static field, method, or property '…'").

**Actual Behavior**:
The second invocation reports error CS1503 ("Argument 1: cannot convert from 'method group' to 'A'") instead of CS0120.

It's unclear to me why the overload resolution prefers the A overload even though a seemingly better overload (MatchesB has a matching signature for the B overload) exists.

Interestingly the error codes switch around if you place the definition of `Method(B)` before `Method(A)` -- it appears as if the overload resolution picks the first candidate in this scenario.

The VB compiler emits BC30518 for an equivalent input and correctly reports that the issue is that an instance member is not accessible in a static/Shared context.

```
Overload resolution failed because no accessible 'Method' can be called with these arguments:
    'Private Shared Sub Method(a As A)': Cannot refer to an instance member of a class from within a shared method or shared member initializer without an explicit instance of the class.
    'Private Shared Sub Method(b As B)': Cannot refer to an instance member of a class from within a shared method or shared member initializer without an explicit instance of the class.
---------
Overload resolution failed because no accessible 'Method' can be called with these arguments:
    'Private Shared Sub Method(a As A)': Method 'Private Sub MatchesB(o As Object)' does not have a signature compatible with delegate 'Delegate Sub A()'.
    'Private Shared Sub Method(b As B)': Cannot refer to an instance member of a class from within a shared method or shared member initializer without an explicit instance of the class.
```
